### DISCUSSION TITLE
I built your CPU

### DISCUSSION THREAD

**Discussion by hillsh** [wuhanstudio/nand2tetris-iverilog#2] [Show and tell]
Hey Wuhanstudio, I built your CPU on a Terasic DE10-Lite FPGA board, and it works as expected.  I plan to extend it to include an Arduino TFT shield for the display output component and a chicklet keyboard for input.  Then I should be able to run some interesting hack programs.  I really enjoyed the NandToTetris courses, they were some of the best online learning I have done. I was inspired to implement the CPU on some hardware and I was glad to find your repository as it short-circuited the process.
Congratulations on your work!

**Comment by wuhanstudio** [comment]
Hi, hillsh,

Congratulations for your achievements, and I'm really happy that this project is helpful.

After going through the Nand2Tetris course on the Java Simulator, I've always been wondering about implementing it on real hardware which seems intimidating for me at first. 

During the lockdown in the U.K., I finally managed to put my hands on it and stick it to the end. It turns out there are not many lines of code to implement it, thus I created this repo, and hope it will help others to jump start.

Looking forward to seeing your implementation of support for Arduino TFT shield.

BTW, there is an interesting project that seems to have implemented LCD, touch and even sound, but it's strange that the source code of the whole project only contains a skeleton. Hope this will help you.

https://gitlab.com/x653/nand2tetris-fpga/

Han
